java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -1246,12 +1248,22 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_makeCudfColumnView(
       ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::STRING}, size, nullptr, valid,
                                       j_null_count, 0, {offsets_column, data_column}));
     } else if (n_type == cudf::type_id::LIST) {
-         JNI_NULL_CHECK(env, j_offset, "offset is null", 0);
-         cudf::column_view *child_view = reinterpret_cast<cudf::column_view *>(j_child);
-         cudf::size_type *offsets = reinterpret_cast<cudf::size_type *>(j_offset);
-         cudf::column_view offsets_column(cudf::data_type{cudf::type_id::INT32}, size + 1, offsets);
-         ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::LIST}, size, nullptr, valid,
+      JNI_NULL_CHECK(env, j_offset, "offset is null", 0);
+      cudf::jni::native_jpointerArray<cudf::column_view> children(env, j_children);
+      cudf::column_view *child_view = reinterpret_cast<cudf::column_view *>(children[0]);
```

Comment:
There should be no reinterpret_cast needed.  The result of the array index should be the correct type already.

I noticed in the implementation of `operator[]` for `native_jpointerArray` there are no bounds checking.  We should either add it in there, or do it here, just in case there is an issue where someone passes in an empty array.

Reply:
Would this meet the empty array check  :  
```
  T *operator[](int index) const {
    if (wrapped.size() == 0) {
      return nullptr;
    }
    return data()[index];
  }

  T *&operator[](int index) {
    if (wrapped.size() == 0) {
      return nullptr;
    }
    return data()[index];
  }
```

Reply:
I am not sure how to change the second call and also if returning nullptr is ok. Should I throw instead?

Reply:
You would want to copy what the regular `native_jArray` does, but you would have to go into wrapped to get access to env. so it is not super simple.  It is probably better to have a follow on issue to fix it properly and for now just do the check in your code. 

```
  N_TYPE operator[](int index) const {
    if (orig == NULL) {
      throw_java_exception(env, NPE_CLASS, "pointer is NULL");
    }
    if (index < 0 || index >= len) {
      throw_java_exception(env, INDEX_OOB_CLASS, "NOT IN BOUNDS");
    }
    return data()[index];
  }

  N_TYPE &operator[](int index) {
    if (orig == NULL) {
      throw_java_exception(env, NPE_CLASS, "pointer is NULL");
    }
    if (index < 0 || index >= len) {
      throw_java_exception(env, INDEX_OOB_CLASS, "NOT IN BOUNDS");
    }
    return data()[index];
  }
```

Reply:
Thanks a lot Bobby. This is informative. I will add a follow on issue, but if I am getting the fix right: I am doing this now : 
```
} else if (n_type == cudf::type_id::LIST) {
      JNI_NULL_CHECK(env, j_offset, "offset is null", 0);
      cudf::jni::native_jpointerArray<cudf::column_view> children(env, j_children);
      JNI_ARG_CHECK(env, (children.size() != 0), "children size is 0", 0);
      cudf::size_type *offsets = reinterpret_cast<cudf::size_type *>(j_offset);
      cudf::column_view offsets_column(cudf::data_type{cudf::type_id::INT32}, size + 1, offsets);
      ret.reset(new cudf::column_view(cudf::data_type{cudf::type_id::LIST}, size, nullptr, valid,
                                                 j_null_count, 0, {offsets_column, *children[0]}));
   }
```
Does this suffice or does 
```
cudf::jni::native_jpointerArray<cudf::column_view> children(env, j_children);
```
yield wrong result when size = 0? Pardon my naivety, JNI is not my forte.

Reply:
That looks fine.